### DISCUSSION TITLE
Using environment variable in the config.toml file

### DISCUSSION THREAD

**Discussion by feekApp** [knqyf263/pet#330] [Q&A] [answered]
Hi,

I was trying to implement a user name independent config file using `$HOME` in the :

```
[General]
    ...
    SnippetFile = "$HOME/.config/pet/snippet.toml" 
    ...
```

But this results in the error: `snippet file not found. $HOME/.config/pet/snippet.toml`, so it seems like I'am not allowed to use environment variables inside the config file. When using the full path for `SnippetFile` everything is working as expected.

A'm I right you can not use environment variables in the config file, of is this a bug. Or do you have any suggestions to make a user independent configuration.

== Feek

**Comment by RamiAwar** [comment, marked as answer]
You're in luck - less than a week ago we merged a PR that supports relative paths 😄 

You can't use environment variables in the values, but you can use relative paths. So in this case, just do `SnippetFile = "~/.config/pet/snippet.toml"` and it should work! Lmk

**Comment by feekApp** [comment]
Great @RamiAwar , that's working,

But now how do I prevent adding a new snippet using `pet new "command --ooption"` in each snippet in `snippet.toml the following item is added:
```
Filename = "/Users/UserName/.config/pet/snippet.toml"
```
So now it is not user name independent anymore.

== Feek

**Reply by RamiAwar** [reply]
I actually think this filename attribute is useless. I need to double check but I created an issue some weeks ago to remove it.

**Reply by feekApp** [reply]
👍

**Comment by feekApp** [comment]
sorry my mistake, the full path was inserted by my previous settings without `~`.... but why are all these `Filename` items inserted, because they are pointing to itself..
== Feek

**Reply by RamiAwar** [reply]
Yup should be removed

**Reply by RamiAwar** [reply]
Already merged this btw - will be there in next release this by weekend!

**Reply by feekApp** [reply]
Thnx for the update today. Works as a charm now! 👍
